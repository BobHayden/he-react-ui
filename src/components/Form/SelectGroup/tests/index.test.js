import createTestContext from 'react-cosmos-test/enzyme'
import fixture from '../SelectGroup.fixture'

const { mount, getWrapper } = createTestContext({ fixture })

beforeEach(mount)

test('<SelectGroup /> rendered correctly with fixture', () => {
  let wrapper = getWrapper()
  Object.keys(fixture.props.options).forEach(key => {
    expect(wrapper.text()).toContain(fixture.props.options[key].label)
  })

  expect(wrapper.find('.select-group-btn')).toHaveLength(Object.keys(fixture.props.options).length)

  expect(wrapper).toMatchSnapshot()
})
