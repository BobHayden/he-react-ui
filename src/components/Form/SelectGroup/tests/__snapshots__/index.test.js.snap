// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectGroup /> rendered correctly with fixture 1`] = `
<SelectGroup
  onClickOption={[Function]}
  options={
    Array [
      Object {
        "label": "General Exam and Clean",
        "value": 1,
      },
      Object {
        "label": "Toothache",
        "value": 2,
      },
      Object {
        "label": "Loose Filling",
        "value": 3,
      },
      Object {
        "label": "Loose Crown",
        "value": 4,
      },
      Object {
        "label": "Teeth Whitening",
        "value": 5,
      },
    ]
  }
  vertical={false}
>
  <styled.div
    className="select-group"
  >
    <div
      className="select-group sc-bdVaJa iCFXPW"
    >
      <div
        className="select-group-btn"
        key="0"
        onClick={[Function]}
      >
        General Exam and Clean
      </div>
      <div
        className="select-group-btn"
        key="1"
        onClick={[Function]}
      >
        Toothache
      </div>
      <div
        className="select-group-btn"
        key="2"
        onClick={[Function]}
      >
        Loose Filling
      </div>
      <div
        className="select-group-btn"
        key="3"
        onClick={[Function]}
      >
        Loose Crown
      </div>
      <div
        className="select-group-btn"
        key="4"
        onClick={[Function]}
      >
        Teeth Whitening
      </div>
    </div>
  </styled.div>
</SelectGroup>
`;
